File: server/Model/Entities/DanhMuc.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Model.Entities;

[Table("DanhMuc")]
public class DanhMuc : EntityBase
{
    [Required(ErrorMessage = "Mã danh mục không được để trống")]
    [MaxLength(100)]
    public string MaDanhMuc { get; set; } = string.Empty;

    [Required(ErrorMessage = "Tên danh mục không được để trống")]
    [MaxLength(250)]
    public string TenDanhMuc { get; set; } = string.Empty;

    public string? MoTa { get; set; }

    [Required(ErrorMessage = "Mã nhóm danh mục không được để trống")]
    [MaxLength(100)]
    public string MaNhomDanhMuc { get; set; } = string.Empty;

    public string? IconUrl { get; set; }
    
    [MaxLength(50)]
    public string? LoaiNgonNgu { get; set; }
    
    public int? ThuTuHienThi { get; set; } = 0;
    
    public bool IsActive { get; set; } = true;
    
    public string? DuongDanFile { get; set; }
    
    public string? UrlLink { get; set; }
}

File: client/src/types/danhMuc.ts
export interface NhomDanhMucDto {
  id: string
  maNhomDanhMuc: string
  tenNhomDanhMuc: string
  iconUrl?: string
  thuTuHienThi?: number
}

export interface NhomDanhMucSearch {
  query?: string
  pageIndex: number
  pageSize: number
  sortColumn?: string
  sortOrder?: string
}

export interface NhomDanhMucCreateVM {
  maNhomDanhMuc: string
  tenNhomDanhMuc: string
  iconUrl?: string
  thuTuHienThi?: number
}

export interface NhomDanhMucEditVM extends NhomDanhMucCreateVM {}

export interface DanhMucDto {
  id: string
  maDanhMuc: string
  tenDanhMuc: string
  moTa?: string
  maNhomDanhMuc: string
  iconUrl?: string
  loaiNgonNgu?: string
  thuTuHienThi?: number
  isActive: boolean
  duongDanFile?: string
  urlLink?: string
}

export interface DanhMucSearch {
  query?: string
  maNhomDanhMuc?: string
  pageIndex: number
  pageSize: number
  sortColumn?: string
  sortOrder?: string
}

export interface DanhMucCreateVM {
  maDanhMuc: string
  tenDanhMuc: string
  moTa?: string
  maNhomDanhMuc: string
  iconUrl?: string
  loaiNgonNgu?: string
  thuTuHienThi?: number
  isActive: boolean
  duongDanFile?: string
  urlLink?: string
}

export interface DanhMucEditVM extends DanhMucCreateVM {}
